indexing/generate_index.py

```diff
@@ -0,0 +1,82 @@
+# TODO: Remove "contents"
+#       Add links within the doc (contents in a hierarchy?)
```

Comment:
Are you still planning on this?

Reply:
Now done

Reply:
When I try the links they are 404-ing - looks like `test_data` is missing in the URL.

Also what would you think of making the heading the link, rather than having separate small text?

Reply:
I did wonder. Also wasn't sure about whether the actual file or the containing folder should be linked, but they can always get back to the folder with the breadcrumbs and it's harder in the other direction. The title should now be the link (though I'll have to run the action to check it all works as expected)

Reply:
Done